src/main/java/net/neoforged/neoforge/event/level/GameRuleChangedEvent.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.level;
+
+import java.util.function.Consumer;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.world.level.gamerules.GameRule;
+import net.minecraft.world.level.gamerules.GameRules;
+import net.neoforged.bus.api.Event;
+import net.neoforged.fml.LogicalSide;
+import net.neoforged.neoforge.common.NeoForge;
+import org.jetbrains.annotations.ApiStatus;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * Event which is fired when ever a {@link GameRule} is updated.
+ * <p>
+ * This event is fired on the {@link NeoForge#EVENT_BUS main NeoForge event bus}, only on both the logical {@linkplain LogicalSide#CLIENT client} and {@linkplain LogicalSide#SERVER}.
+ */
+public final class GameRuleChangedEvent extends Event {
```

Comment:
Is there a particular reason why the event is fired from `GameRules#set()` instead of `MinecraftServer#onGameRuleChanged()`? I can't think of a use case where one would want to listen to updates when there is no server and the fact that the server can be null even if there is a server also makes proper handling of the logical side a complete nightmare.